src/util.cpp

```diff
@@ -1211,619 +1211,656 @@ CInstPictures::EInstCategory CInstPictures::GetCategory ( const int iInstrument
 }
 
 // Locale management class -----------------------------------------------------
-QString CLocale::GetCountryFlagIconsResourceReference ( const QLocale::Country eCountry )
+QLocale::Country CLocale::WireFormatCountryCodeToQtCountry ( unsigned short eCountryCode )
+{
+#if QT_VERSION >= QT_VERSION_CHECK( 6, 0, 0 )
+    // The Jamulus protocol wire format gives us Qt5 country IDs.
+    // Qt6 changed those IDs, so we have to convert back:
+    return (QLocale::Country) qt5CountryToQt6Table[eCountryCode];
+#else
+    return (QLocale::Country) eCountryCode;
+#endif
+}
+
+unsigned short CLocale::QtCountryToWireFormatCountryCode ( const QLocale::Country eCountry )
+{
+#if QT_VERSION >= QT_VERSION_CHECK( 6, 0, 0 )
+    // The Jamulus protocol wire format expects Qt5 country IDs.
+    // Qt6 changed those IDs, so we have to convert back:
+    return qt6CountryToQt5Table[(unsigned short) eCountry];
+#else
+    return (unsigned short) eCountry;
+#endif
+}
+
+bool CLocale::IsCountryCodeSupported ( unsigned short usCountryCode )
+{
+#if QT_VERSION >= QT_VERSION_CHECK( 6, 0, 0 )
+    // On newer Qt versions there might be codes which do not have a Qt5 equivalent.
+    // We have no way to support those sanely right now.
+    return qt6CountryToQt5Table[usCountryCode] != -1;
+#else
+    // All Qt5 codes are supported.
+    Q_UNUSED ( usCountryCode );
+    return true;
+#endif
+}
+
+QString CLocale::GetCountryFlagIconsResourceReference ( const QLocale::Country eCountry /* Qt-native value */ )
 {
     QString strReturn = "";
 
     // special flag for none
     if ( eCountry == QLocale::AnyCountry )
     {
-        strReturn = ":/png/flags/res/flags/flagnone.png";
+        return ":/png/flags/res/flags/flagnone.png";
     }
-    else
+    // NOTE: The following code was introduced to support old QT versions. The problem
+    //       is that the number of countries displayed is less than the one displayed
+    //       with the new code below (which is disabled). Therefore, as soon as the
+    //       compatibility to the very old versions of QT is not required anymore, use
+    //       the new code.
+    // COMPATIBLE FOR OLD QT VERSIONS -> use a table:
+    QString strISO3166   = "";
+    uint    iCountryCode = static_cast<uint> ( eCountry );
```

Comment:
So here we're using a `uint` (not `unsigned short`) and calling the variable `i...` (without a `u`)...

Hmm!  Not consistent.